libraries/AP_OpticalFlow/AP_OpticalFlow_Linux.cpp

```diff
@@ -0,0 +1,154 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ *       AP_OpticalFlow_Linux.cpp - ardupilot library for the PX4Flow sensor.
+ *          inspired by the PX4Firmware code.
+ *      
+ *       @author: Víctor Mayoral Vilches
+ *
+ */
+
+#include <AP_HAL.h>
+#include "OpticalFlow.h"
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+extern const AP_HAL::HAL& hal;
+
+AP_OpticalFlow_Linux::AP_OpticalFlow_Linux(OpticalFlow &_frontend) : 
+  OpticalFlow_backend(_frontend)
+{}
+
+
+void AP_OpticalFlow_Linux::init(void)
+{
+  uint8_t buff[22];
+
+  // get pointer to i2c bus semaphore
+  AP_HAL::Semaphore *i2c_sem = hal.i2c->get_semaphore();
+
+  // take i2c bus sempahore
+  if (!i2c_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+      hal.scheduler->panic(PSTR("PX4FLOW: unable to get semaphore"));
+  }
+
+  // to be sure this is not a ll40ls Lidar (which can also be on
+  // 0x42) we check if a I2C_FRAME_SIZE byte transfer works from address
+  // 0. The ll40ls gives an error for that, whereas the flow
+  // happily returns some data
+  uint8_t val[I2C_FRAME_SIZE];
+  if (hal.i2c->readRegisters(I2C_FLOW_ADDRESS, 0, I2C_FRAME_SIZE, val))
+    hal.scheduler->panic(PSTR("ll40ls Lidar"));
```

Comment:
we should not be using panic() here. Instead set a flag saying it hasn't found the sensor and then fail the read()
All of the panic() calls should be removed from this code